src/System.Diagnostics.StackTrace.Symbols/src/System/Diagnostics/StackTrace/Symbols.cs

```diff
@@ -0,0 +1,222 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Reflection.PortableExecutable;
+
+namespace System.Diagnostics.StackTrace
+{
+    public static class Symbols
+    {
+        /// <summary>
+        /// Returns the source file and line number information for the method.
+        /// </summary>
+        /// <param name="assemblyPath">file path of the assembly or null</param>
+        /// <param name="inMemorySymbols">in memory PDB address or zero</param>
+        /// <param name="inMemorySymbolsSize">in memory PDB size</param>
+        /// <param name="methodToken">method token</param>
+        /// <param name="ilOffset">il offset of the stack frame</param>
+        /// <param name="sourceFile">source file return</param>
+        /// <param name="sourceLine">line number return</param>
+        /// <param name="sourceColumn">column return</param>
+        public static void GetSourceLineInfo(string assemblyPath, IntPtr inMemorySymbols, int inMemorySymbolsSize, 
+            int methodToken, int ilOffset, out string sourceFile, out int sourceLine, out int sourceColumn)
+        {
+            sourceFile = null;
+            sourceLine = 0;
+            sourceColumn = 0;
+
+            MetadataReader reader;
+            using (GetReader(assemblyPath, inMemorySymbols, inMemorySymbolsSize, out reader)) {
+                if (reader != null) {
```

Comment:
Personal style preference, I find it easier to read when the code doesn't need so many levels of indentation:

    if(reader == null)
    {
        return;
    }
    ...